backend/app/controllers/projects.js

```diff
@@ -384,21 +397,33 @@ module.exports = {
     userRemoval: function (req, res, next) {
         var thunkQuery = req.thunkQuery;
         co(function* () {
-            yield thunkQuery(
-                'DELETE FROM "ProjectUsers" WHERE "ProjectUsers"."projectId" = ' +
-                req.params.id + ' AND "ProjectUsers"."userId" = ' + req.params.userId
-            );
+            var projectExist = yield * common.checkRecordExistById(req, 'ProjectUsers', 'projectId', req.params.projectId);
+            var userExist = yield * common.checkRecordExistById(req, 'ProjectUsers', 'userId', req.params.userId);
+
+            if (projectExist === true && userExist === true) {
 
-            var productId = _.first(_.map((yield thunkQuery(
-                Product.select(Product.id).from(Product).where(Product.projectId.equals(req.params.id))
-            )), 'id'));
+                yield thunkQuery(
+                    'DELETE FROM "ProjectUsers" WHERE "ProjectUsers"."projectId" = ' +
+                    req.params.projectId + ' AND "ProjectUsers"."userId" = ' + req.params.userId
+                );
 
-            yield thunkQuery(
-                'DELETE FROM "Tasks" WHERE "Tasks"."productId" = ' + productId +
-                ' AND ' + req.params.userId + ' = ANY("Tasks"."userIds")'
-            )
+                var productId = _.first(_.map((yield thunkQuery(
+                    Product.select(Product.id).from(Product).where(Product.projectId.equals(req.params.id))
+                )), 'id'));
 
-            return true;
+                if (productId) {
+                    return yield thunkQuery(
+                        'UPDATE "Tasks" ' +
+                        'SET "Tasks"."isDeleted" = '+ Date.now() +
+                        'WHERE "Tasks"."productId" = ' + productId +
+                        ' AND ' + req.params.userId + ' = ANY("Tasks"."userIds")'
```

Comment:
This still needs to be tested to make sure that it actually marks it as deleted. Another test which is failing is preventing me from doing this.